sdk/monitor/azure-monitor-opentelemetry-exporter/src/main/java/com/azure/monitor/opentelemetry/exporter/implementation/logging/DiagnosticTelemetryPipelineListener.java

```diff
@@ -117,23 +116,24 @@ public CompletableResultCode shutdown() {
 
     public static String getErrorMessageFromCredentialRelatedResponse(
         int responseCode, String responseBody) {
-        JsonNode jsonNode;
-        try {
-            jsonNode = new ObjectMapper().readTree(responseBody);
-        } catch (JsonProcessingException e) {
+        String message = null;
+        try (JsonReader jsonReader = JsonProviders.createReader(responseBody)) {
+            Response response = Response.fromJson(jsonReader);
+            List<ResponseError> responseErrorList = response.getErrors().stream()
+                .filter(error -> error.getStatusCode() == responseCode).collect(Collectors.toList());
```

Comment:
This looks to be a change in behavior where the previous design didn't filter the errors on their status code. Is this intentional?

Reply:
this is new and yes it's intentional. we need to handle 206 partial response code.. that is why we didn't have response and reponseError models. now we do (Response.java and ResponseError.java under models).

Reply:
updated code to support newlines deserialization https://github.com/Azure/azure-sdk-for-java/pull/41106/commits/c9502466e69f264d925c7692d055c4291afc5638

Reply:
@heyams is the idea here that the top-level response status code always has at least one response item with the same status code? or do we just want to filter out `206` status code?

if it's the former, can you add a comment in the code to mention? thanks

Reply:
when it's 200, errors is empty, when 400 returned for the whole batch, it would have one error return. for the case of 206, it will an array of errors returned. does it help? let's chat if not.